FirebaseTestProject/Models/Post.swift

```diff
@@ -49,13 +79,37 @@ struct PostService {
         return db.collection("posts")
     }
     
+    static var favoritesReference: CollectionReference {
+        let db = Firestore.firestore()
+        return db.collection("favorites")
+    }
+    
     static func getPosts() async throws -> [Post] {
         let postsSnapshots = try await postsReference.getDocuments()
         let posts = postsSnapshots.documents.map { Post(from: $0.data()) }
         return posts
     }
     
+    static func getFavorites() async throws -> [Post] {
+        let userid = UserDefaults.standard.value(forKey: "userid") != nil ?  UserDefaults.standard.value(forKey: "userid") as! String : "00854E9E-8468-421D-8AA2-605D8E6C61D9"
+        let favoritesQuery = favoritesReference.whereField("userid", isEqualTo: userid)
+        let favoritesSnapshots = try await favoritesQuery.getDocuments()
+        let favorites = favoritesSnapshots.documents.map { Post(from: $0.data()) }
+        
+        return favorites
+    }
+    
     static func upload(_ post: Post) async throws {
         try await postsReference.document(post.id.uuidString).setData(post.jsonDict)
     }
+    
+    static func favorite(_ post: Post) async throws {
+        let userid = UserDefaults.standard.value(forKey: "userid") as? String ?? "00854E9E-8468-421D-8AA2-605D8E6C61D9"
```

Comment:
Please see the note above regarding retrieving the `userid` in this way.

Reply:
LMK: Generally speaking, we are going to have to save a UserID during authentication, though. And, we would only ever need to get a favorite for the current user. I would think that we would automatically call userID inside the function since it isn't ever going to change per a parameter.

Reply:
Be it UserDefaults or some other method, I just assumed that we would need to store the value. That could be a false assumption.